### PR TITLE
Change 0.3.2

### PR DESCRIPTION
**Pull request by FynnBe**
changes as summarized in https://github.com/bioimage-io/spec-bioimage-io/issues/91#issuecomment-854905623

remaining problem: auto conversion for `documentation`:
- [ ] converting an invalid input (e.g. a url) to a local md file requires the __whole package__, not just the yaml content
    - not a problem for 'validate-spec' (<- should be renamed 'validate-model'.?), which takes a model.yaml path, but a problem for 'validate-bioimageio-manifest' (should be renamed 'validate-manifest'?)
 - [ ] requires python packager to create package from yaml url

### CODE REVIEW

**Comment by FynnBe** [comment]
relates to #98

File: bioimageio/spec/__main__.py [hunk 1]
```diff
@@ -14,19 +14,24 @@
 app = typer.Typer()  # https://typer.tiangolo.com/
 
 
+@app.command()
+def package_model(model_yaml: Path, auto_convert: bool = False):
```
Comment (constantinpape) [inline bioimageio/spec/__main__.py:18]: General comment: Can we add docstrings to the CLI? I think it would be good to get something meaningful on `package_model --help` etc.
(Fine to do in a separate PR, just wanted to point this out.)

File: bioimageio/spec/utils/maybe_convert_to_v0_3.py [hunk 2]
```diff
@@ -154,11 +153,19 @@ def convert_model_v0_3_1_to_v0_3_2(data: Dict[str, Any]) -> Dict[str, Any]:
...
+        if authors_update is not None:
+            for a, u in zip(weights_entry["authors"], authors_update):
+                a.update(u)
+
+    # documentation (if not a local relative md file, pretend content is in local 'auto_converted_documentation.md')
+    class DocSchema(Schema):
+        doc = schema.Model.documentation
+
+    if DocSchema().validate({"doc": data["documentation"]}):
+        data["config"] = data.get("config", {})  # make sure config exists
```
Comment (constantinpape) [inline bioimageio/spec/utils/maybe_convert_to_v0_3.py:165]: Why is "config" involved here? I'd assume that we just put the content of the "documentation" field into `AUTO_CONVERTED_DOCUMENTATION_FILE_NAME`?

**Comment by constantinpape** [comment]
> changes as summarized in [#91 (comment)](https://github.com/bioimage-io/spec-bioimage-io/issues/91#issuecomment-854905623)
> 
> remaining problem: auto conversion for `documentation`:
> * converting an invalid input (e.g. a url) to a local md file requires the **whole package**, not just the yaml content
>   
>   * not a problem for 'validate-spec' (<- should be renamed 'validate-model'.?), which takes a model.yaml path, but a problem for 'validate-bioimageio-manifest' (should be renamed 'validate-manifest'?)
> 

If I understand this right, the problem is that if we only have the yaml we cannot fully auto convert with missing documentation.
But if we only have the yaml we cannot fully validate the package in any case, because, for example, we can't check if the .md file exists.
So shouldn't it be fine to just change the documentation field and pretend that the auto converted .md file would exist?

File: bioimageio/spec/utils/maybe_convert_to_v0_3.py (hunk 2 above)
Comment (FynnBe) [reply bioimageio/spec/utils/maybe_convert_to_v0_3.py:165]: that would not pass the schema, so I thought `config` would be a good place too keep it.

File: bioimageio/spec/__main__.py (hunk 1 above)
Comment (FynnBe) [reply bioimageio/spec/__main__.py:18]: totally! 👍

**Comment by FynnBe** [comment]
> So shouldn't it be fine to just change the documentation field and pretend that the auto converted .md file would exist?

yes, that's what this PR does so far...

File: bioimageio/spec/utils/maybe_convert_to_v0_3.py (hunk 2 above)
Comment (constantinpape) [reply bioimageio/spec/utils/maybe_convert_to_v0_3.py:165]: Yes, but why are you adding 
```python
data["config"][AUTO_CONVERTED_DOCUMENTATION_FILE_NAME] = data["documentation"]
```
in the first place?
Is it just to keep track of the fact that the documentation has been auto generated?

Comment (FynnBe) [reply bioimageio/spec/utils/maybe_convert_to_v0_3.py:165]: to save the original content of `data["documentation"]` (that would need to be written to the AUTO_CONVERTED_DOCUMENTATION_FILE_NAME file)

Comment (FynnBe) [reply bioimageio/spec/utils/maybe_convert_to_v0_3.py:165]: as the converter only has the yaml content as dict it cannot write the file as it wouldn't know where to.

Comment (constantinpape) [reply bioimageio/spec/utils/maybe_convert_to_v0_3.py:165]: Ok, makes sense. Could you sum this up in a comment though? This is not obvious from the code alone.

**Comment by constantinpape** [comment]
@FynnBe should we go ahead and merge this, I think all the changes are approved now and the test issues are only due to https://github.com/deepimagej/models/pull/22.

**Comment by FynnBe** [comment]
the following todos can be left for future PRs:
- [ ] reactive ci for tests (potentially with #103)
- [ ] clean up `schema.Model` base class `schema.RDF` (mostly just move around fields)
- [ ] CLI to verify models and manifest for specific versions (and not juts the latest)
- [ ]  add more example specs here?

**Comment by oeway** [comment]
Would it make more sense to make it `0.4.0` instead?

**Comment by FynnBe** [comment]
> Would it make more sense to make it 0.4.0 instead?

the stricter requirements for `license` are the only thing that could break a 0.3.1 model. It would be the cleanest to go 0.4.0, but I think it's not worth the effort of changing this now, as that part is already in master.

if this really is an issue I would change it to only warn about the license, but we do not have an established "warn about deprecattion" test/validator/tool, which we should add for better transitions to future versions

**Comment by constantinpape** [comment]
Yeah, I think we should have been more careful when doing the initial bump to 0.3.2 and named it 0.4.0 already.
I think the changes here are just to fix some of the inconsistencies we had when bumping to 0.3.2 so we should merge it without complicating it further.

**Review by constantinpape — APPROVED** [review]

**Comment by constantinpape** [comment]
Let's also move the  checklist from https://github.com/bioimage-io/spec-bioimage-io/pull/97#issuecomment-859533766 to some issue once this is merged so that we can keep track of it.

**Comment by FynnBe** [comment]
> Let's also move the checklist from #97 (comment) to some issue once this is merged so that we can keep track of it.

created #105
